corehq/util/es/interface.py

```diff
@@ -20,12 +20,8 @@ def put_mapping(self, doc_type, mapping, index):
         return self.es.indices.put_mapping(doc_type, {doc_type: mapping}, index=index)
 
     def _verify_is_alias(self, index_or_alias):
-        from corehq.elastic import ES_META, ESError
-        from pillowtop.tests.utils import TEST_ES_ALIAS
-        all_es_aliases = [index_info.alias for index_info in ES_META.values()] + [TEST_ES_ALIAS]
-        if index_or_alias not in all_es_aliases:
-            raise ESError(
-                f"{index_or_alias} is an unknown alias, query target must be one of {all_es_aliases}")
+        from corehq.elastic import verify_registered_alias
+        verify_registered_alias(index_or_alias)
```

Comment:
I went back and forth on this when I wrote it. In the end I decided not to add a cache because "if it was good enough before...".  Also, I wrote this before I fully understood what this entire mechanism was for (why we still need it).  I forged forward with the commits on this PR as to not block further progress on async communication (the current implementation is not worse than what it replaces), but there is a good chance we can rip this entire registry logic out and not perform any of these checks at all.